src/Lumi/Components/Form/Table.purs

```diff
@@ -0,0 +1,272 @@
+module Lumi.Components.Form.Table
+  ( TableFormBuilder
+  , column
+  , column_
+  , editableTable
+  , nonEmptyEditableTable
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Array.NonEmpty as NEA
+import Data.Either (Either(..))
+import Data.Foldable (foldMap, intercalate)
+import Data.FunctorWithIndex (mapWithIndex)
+import Data.Lens.Index (ix)
+import Data.Maybe (Maybe, fromMaybe, isNothing, maybe)
+import Data.Monoid (guard)
+import Data.Newtype (class Newtype)
+import Data.Nullable as Nullable
+import Data.Traversable (traverse)
+import Data.Tuple (Tuple(..))
+import Effect (Effect)
+import Lumi.Components.Column as Column
+import Lumi.Components.EditableTable as EditableTable
+import Lumi.Components.Form.Internal (FormBuilder(..), Tree(..), Forest, formBuilder)
+import Lumi.Components.LabeledField (ValidationMessage(..))
+import Lumi.Components.Orientation (Orientation(..))
+import Lumi.Components.Row as Row
+import Lumi.Components.Spacing (Space(..), hspace, vspace)
+import Lumi.Components.Text as T
+import React.Basic (JSX, keyed)
+import React.Basic.DOM as R
+import Unsafe.Reference (unsafeRefEq)
+
+-- | An applicative functor used to build editable tables out of `FormBuilder`s.
+-- | `FormBuilder`s can be turned into `TableFormBuilder`s with the `column`
+-- | function. `TableFormBuilder`s can, then, be used inside forms with the
+-- | `editableTable` function.
+newtype TableFormBuilder props row result =
```

Comment:
Is there a reason not to move this down to the `EditableTable` module and use it in `EditableTable.editableTable`? It might be useful outside of (bigger) forms...

Reply:
Hmm, yes, that's a good idea. We'd still need to rely on `Form.build` to actually use it outside of forms though, so I'm not sure if its place in the hierarchy is outside of `Lumi.Compnents.Form.*`.

Reply:
I'm fine with this either way actually.